### PR TITLE
feat: [NOD-782] [dismissione WISP] Add APIM policy for caching keys for decoupler

### PR DESCRIPTION
**Pull request by andrea-deri**
⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️
⚠️⚠️ resolve `# PSFC-TODO` before merge 
⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️

### List of changes

- Add api APIM to cache WISP Info `WISP Converter - Caching`  **POST** `/save-mapping`
- Adapt decoupler apim policy for WISP
- and other stuffs ... 

to apply 

```
shared-common
yes yes | sh terraform.sh apply weu-<ENV>

cd domains/nodo-app
yes yes | sh terraform.sh apply weu-<ENV>

gps-app
yes yes | sh terraform.sh apply weu-<ENV>

core
yes yes | sh terraform.sh apply <ENV>
```

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/nodo-app/04_apim_wisp_converter.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
+##############
+## Products ##
+##############
+module "apim_wisp_converter_product" {
```
Comment (aomegax) [inline src/domains/nodo-app/04_apim_wisp_converter.tf:4]: for each resources please add `var.enable_wisp_converter` to release in specific env

File: src/domains/nodo-app/04_apim_wisp_converter.tf [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  resource_group_name   = local.pagopa_apim_rg
+  product_ids           = [module.apim_wisp_converter_product.product_id]
+  subscription_required = true
+
+  version_set_id = azurerm_api_management_api_version_set.api_version_set_wisp_converter_decoupler_caching.id
+  api_version    = "v1"
+
+  description  = "API for caching data for decoupler in WISP Converter"
+  display_name = "WISP Converter - Decoupler caching"
+  path         = "wisp-converter-decoupler/api"
```
Comment (aomegax) [inline src/domains/nodo-app/04_apim_wisp_converter.tf:56]: if we use the following convention:
- `wisp-converter/rest`
- `wisp-converter/soap`
- `wisp-converter/decoupler`
 
what do you think about?

File: src/domains/nodo-app/04_apim_wisp_converter.tf [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  protocols    = ["https"]
+
+  service_url = null
+
+  content_format = "openapi"
+  content_value = templatefile("./api/wisp-converter-decoupler-caching/v1/openapi.json", {
```
Comment (aomegax) [inline src/domains/nodo-app/04_apim_wisp_converter.tf:62]: we should use `./api/wisp-converter` (like `api_product`) as main folder and then `/decoupler`, `/rest` and so on.

File: src/domains/nodo-app/api/wisp-converter-decoupler-caching/v1/_base_policy.xml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+<policies>
+	<inbound>
+		<base />
+		<set-variable name="nav_keys" value="@((JArray)context.Request.Body.As<JObject>()["keys"])" />
```
Comment (aomegax) [inline src/domains/nodo-app/api/wisp-converter-decoupler-caching/v1/_base_policy.xml:4]: to review business logic

in general use policy fragment to generalize code

**Review by aomegax — CHANGES_REQUESTED** [review]

File: src/domains/nodo-app/04_apim_wisp_converter.tf (hunk 2 above)
Comment (andrea-deri) [reply src/domains/nodo-app/04_apim_wisp_converter.tf]: Good idea! I will change the paths as described.

**Pull request merged**